File: LDC.Atlas.WebApp/Client/app/shared/services/execution/dtos/transaction-report-command.ts
import { BalancesType } from '../../../../shared/enums/balances-type.enum';
import { MatchingsType } from '../../../../shared/enums/matchings-type.enum';
import { NominalAccountType } from '../../../../shared/enums/nominal-account-type-enum';
import { ReportStyleType } from '../../../../shared/enums/report-style-type.enum';
import { UnmatchedType } from '../../../../shared/enums/unmatched-type.enum';

export class TransactionReportCommand {
    fromDate: Date;
    toDate: Date;
    balanceType: BalancesType;
    reportStyleType: ReportStyleType;
    matchingType: MatchingsType;
    unmatchedType: UnmatchedType;
    functionalCurrency: boolean;
    broughtForward :boolean;
    accuralsIncluded: boolean;
    accountType: NominalAccountType;
    documentFromDate: Date;
    documentToDate: Date;
}

File: LDC.Atlas.Services.PreAccounting/Application/Commands/TransactionReportCommand.cs
using LDC.Atlas.Services.PreAccounting.Entities;
using System;

namespace LDC.Atlas.Services.PreAccounting.Application.Commands
{
    public class TransactionReportCommand
    {
        public DateTime? FromDate { get; set; }

        public DateTime? ToDate { get; set; }

        public BalancesType? BalanceType { get; set; }

        public ReportStyleType? ReportStyleType { get; set; }

        public MatchingsType? MatchingType { get; set; }

        public UnmatchedType? UnmatchedType { get; set; }

        public bool? FunctionalCurrency { get; set; }

        public bool BroughtForward { get; set; }

        public bool? AccuralsIncluded { get; set; }

        public NominalAccountType? AccountType { get; set; }

        public DateTime? DocumentFromDate { get; set; }

        public DateTime? DocumentToDate { get; set; }
    }
}
